src/UI/Component/Layout/Page/Cockpit/Factory.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace ILIAS\UI\Component\Layout\Page\Cockpit;
+/**
+ * This is the factory for cockpit-components
+ */
+interface Factory {
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The bar is the primary visual manifestation of the cockpit.
+     *
+     *   composition: >
+     *     The bar is the cockpit's contanier for entries.
+     *
+     *   effect: >
+     *     In a desktop environment, a vertical bar is rendered on the left side
+     *     of the screen covering the full height minus the header area.
+     *     Entries are aligned vertically.
+     *
+     *     For mobile devices, the bar is rendered horizontally on the bottom
+     *     of the screen with the entries aligned horizontally.
+     *
+     *     The cockpit(-bar) is always visible and available (except in exam/kiosk mode).
+     *
+     * rules:
+     *   usage:
+     *     1: There MUST be but one cockpit bar on the page.
+     *
+     *   composition:
+     *     1: The bar MUST NOT contain items other than entries.
+     *     2: The bar MUST contain at least one entry.
+     *     3: The bar MUST NOT contain more than five entries.
+     *
+     *   style:
+     *     1: The bar MUST have a fixed witdth (desktop).
+     *     2: The bar MUST have a fixed height (mobile).
+     *
+     * ----
+     * @param  entry[] $entries
+     * @return  \ILIAS\UI\Component\Layout\Page\Cockpit\Bar
+     */
+    public function bar($entry_1, $entry_2=null, $entry_3=null, $entry_4=null, $entry_5=null);
+
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     An entry in this context describes an item in the cockpit-bar.
+     *     Usually, it is a labeled, clickable icon which triggers a slate.
+     *     Alternatively, the link directly targets a page.
+     *
+     *   composition: >
+     *     Entries consist of a caption, an icon and, optionally, a slate.
+     *
+     * rules:
+     *   usage:
+     *     1: Entries MUST only be inside a cockpit-bar.
+     *     2: Entries CAN be connected with a slate
+     *
+     *   interaction:
+     *     1: Entries MUST have an active-state.
```

Comment:
+1. Please add some indication, at which point this active-state should be active. This can be quite hard. E.g. which (if any) entry is active by looking at some blog inside the repository. Which is active while looking at the administration? Which is active while looking at the PD? Are they only active if the slate is open?